Makefile

```diff
@@ -11,127 +11,9 @@ endif
-EFI_LDS		= $(TOPDIR)/elf_$(ARCH)_efi.lds
-
-CFLAGS		= -ggdb -O0 -fno-stack-protector -fno-strict-aliasing -fpic \
-		  -fshort-wchar -Wall -Wsign-compare -Werror -fno-builtin \
-		  -Werror=sign-compare -ffreestanding -std=gnu89 \
-		  -I$(shell $(CC) -print-file-name=include) \
-		  "-DDEFAULT_LOADER=L\"$(DEFAULT_LOADER)\"" \
-		  "-DDEFAULT_LOADER_CHAR=\"$(DEFAULT_LOADER)\"" \
-		  $(EFI_INCLUDES)
-
-COMMITID ?= $(shell if [ -d .git ] ; then git log -1 --pretty=format:%H ; elif [ -f commit ]; then cat commit ; else echo commit id not available; fi)
-
-ifneq ($(origin OVERRIDE_SECURITY_POLICY), undefined)
-	CFLAGS	+= -DOVERRIDE_SECURITY_POLICY
-endif
-
-ifneq ($(origin ENABLE_HTTPBOOT), undefined)
-	CFLAGS	+= -DENABLE_HTTPBOOT
-endif
-
-ifeq ($(ARCH),x86_64)
-	CFLAGS	+= -mno-mmx -mno-sse -mno-red-zone -nostdinc \
-		   -maccumulate-outgoing-args \
-		   -DEFI_FUNCTION_WRAPPER -DGNU_EFI_USE_MS_ABI \
-		   -DNO_BUILTIN_VA_FUNCS -DMDE_CPU_X64 -DPAGE_SIZE=4096
-	LIBDIR			?= $(prefix)/lib64
-	ARCH_SUFFIX		?= x64
-	ARCH_SUFFIX_UPPER	?= X64
-	ARCH_LDFLAGS		?=
-endif
-ifeq ($(ARCH),ia32)
-	CFLAGS	+= -mno-mmx -mno-sse -mno-red-zone -nostdinc \
-		   -maccumulate-outgoing-args -m32 \
-		   -DMDE_CPU_IA32 -DPAGE_SIZE=4096
-	LIBDIR			?= $(prefix)/lib
-	ARCH_SUFFIX		?= ia32
-	ARCH_SUFFIX_UPPER	?= IA32
-	ARCH_LDFLAGS		?=
-endif
-ifeq ($(ARCH),aarch64)
-	CFLAGS += -DMDE_CPU_AARCH64 -DPAGE_SIZE=4096 -mstrict-align
-	LIBDIR			?= $(prefix)/lib64
-	ARCH_SUFFIX		?= aa64
-	ARCH_SUFFIX_UPPER	?= AA64
-	FORMAT			:= -O binary
-	SUBSYSTEM		:= 0xa
-	ARCH_LDFLAGS		+= --defsym=EFI_SUBSYSTEM=$(SUBSYSTEM)
-endif
-ifeq ($(ARCH),arm)
-	CFLAGS += -DMDE_CPU_ARM -DPAGE_SIZE=4096 -mstrict-align
-	LIBDIR			?= $(prefix)/lib
-	ARCH_SUFFIX		?= arm
-	ARCH_SUFFIX_UPPER	?= ARM
-	FORMAT			:= -O binary
-	SUBSYSTEM		:= 0xa
-	ARCH_LDFLAGS		+= --defsym=EFI_SUBSYSTEM=$(SUBSYSTEM)
-endif
-
-FORMAT		?= --target efi-app-$(ARCH)
-EFI_PATH	?= $(LIBDIR)/gnuefi
-
-MMSTEM		?= mm$(ARCH_SUFFIX)
-MMNAME		= $(MMSTEM).efi
-MMSONAME	= $(MMSTEM).so
-FBSTEM		?= fb$(ARCH_SUFFIX)
-FBNAME		= $(FBSTEM).efi
-FBSONAME	= $(FBSTEM).so
-SHIMSTEM	?= shim$(ARCH_SUFFIX)
-SHIMNAME	= $(SHIMSTEM).efi
-SHIMSONAME	= $(SHIMSTEM).so
-SHIMHASHNAME	= $(SHIMSTEM).hash
-BOOTEFINAME	?= BOOT$(ARCH_SUFFIX_UPPER).EFI
-BOOTCSVNAME	?= BOOT$(ARCH_SUFFIX_UPPER).CSV
-
-CFLAGS += "-DEFI_ARCH=L\"$(ARCH_SUFFIX)\"" "-DDEBUGDIR=L\"/usr/lib/debug/usr/share/shim/$(ARCH_SUFFIX)-$(VERSION)$(DASHRELEASE)/\""
-
-ifneq ($(origin VENDOR_CERT_FILE), undefined)
-	CFLAGS += -DVENDOR_CERT_FILE=\"$(VENDOR_CERT_FILE)\"
-endif
-ifneq ($(origin VENDOR_DBX_FILE), undefined)
-	CFLAGS += -DVENDOR_DBX_FILE=\"$(VENDOR_DBX_FILE)\"
-endif
-
-LDFLAGS		= --hash-style=sysv -nostdlib -znocombreloc -T $(EFI_LDS) -shared -Bsymbolic -L$(EFI_PATH) -L$(LIBDIR) -LCryptlib -LCryptlib/OpenSSL $(EFI_CRT_OBJS) --build-id=sha1 $(ARCH_LDFLAGS)
+include $(TOPDIR)/Make.defaults
+include $(TOPDIR)/Make.rules
+include $(TOPDIR)/Make.coverity
```

Comment:
... but it doesn't do the same for Make.coverity, so this will leads to the following error:

```
$ make -j5
Makefile:16: /home/javier/devel/shim/Make.coverity: No such file or directory
make: *** No rule to make target '/home/javier/devel/shim/Make.coverity'.  Stop.
```

But I think that's better to postpone adding these empty files and instead add the files and includes in the next commit.